resources/assets/js/annotations/ol/PolygonValidator.js

```diff
@@ -53,9 +52,9 @@ export function makePolygonSimple(feature) {
         return feature;
     }
 
-    // Divide non-simple polygon at cross points into several smaller polygons
-    // and translate back to ol geometry
-    let polygons = jstsValidate(jstsPolygon)['array'].map(p => parser.write(p));
+    // Divide non-simple polygon at cross points into several smaller polygons,
+    // translate back to ol geometry and polish them
+    let polygons = polishPolygons(jstsValidate(jstsPolygon).array.map(p => parser.write(p)));
```

Comment:
It's not immediately clear what "polishing" means. I'd at least expect the comment here to explain a little but it also just says "polish".